libraries/provider_helpers.rb

```diff
@@ -80,7 +96,7 @@ def validate_versions # rubocop:disable Metrics/PerceivedComplexity, Metrics/Cyc
         end
 
         # Check that the package's version is not a pre-release when we really expect a release
-        if !app_version.prerelease? && pkg_version.prerelease?
+        if current_resource.version && !current_resource.version.prerelease? && pkg_version.prerelease?
```

Comment:
For my clarity... `current_resource.version` = installed version.... `app_version` = desired version ... and `pkg_version`  = the version that is exactly installed...

I'm trying to remember but I don't know that this check is actually valid... 

going from "1.0.0" -> "2.0.0-alpha" is totally valid... going "1.0.0" -> "1.0.0-alpha" is not valid, but if 1.0.0 is installed, and 1.0.0 is the desired version, we shouldn't be downloading the package anyways...

Reply:
05dd3b618685d1d409e9730a84d78322933ca46f